noxfile.py

```diff
@@ -45,122 +43,62 @@
-    category: Literal["docs", "tests", "all"],
-    version: Literal["3.11", "3.12", "3.13"],
-    resolution: Literal["highest", "lowest-direct", "lowest"] = "highest",
-    os_platform: Literal["linux", "macos", "windows"] | None = None,
-) -> str:
-    """
-    Return the file path to the requirements file.
+uv_sync = ("uv", "sync", "--no-progress", "--frozen")
 
-    Parameters
-    ----------
-    category : str
-        The name of the optional dependency set, as defined in
-        :file:`pyproject.toml`.
+running_on_ci = os.getenv("CI")
+running_on_rtd = os.environ.get("READTHEDOCS") == "True"
 
-    version : str
-        The supported version of Python.
+uv_requirement = "uv >= 0.6.1"
 
-    resolution : str
-        The resolution strategy used by uv.
 
-    os_platform : str, optional
-        The name of the target platform. By default, it will attempt to find the
-        requirement file associated with the current platform.
+@nox.session
+def requirements(session) -> None:
     """
+    Regenerate the uv.lock file for running tests and building the
+    documentation.
+    """
+    session.install(uv_requirement)
 
-    if os_platform is None:
-        current_platform = platform.system().lower()
-        os_platform = (
-            current_platform
-            if current_platform in supported_operating_systems
-            else "linux"
-        )
-
-    requirements_directory = "ci_requirements"
-    specifiers = [category, version, os_platform]
-    if resolution != "highest":
-        specifiers.append(resolution)
-    return f"{requirements_directory}/{'-'.join(specifiers)}.txt"
+    # If it becomes possible to exclude the current project when using
+    # `uv lock`, we should do so here. That would allow us to add a Nox
+    # session to validate the requirements back in.
+    session.run("uv", "lock", "--upgrade", "--no-progress")
 
 
 @nox.session
-def requirements(session) -> None:
+def validate_requirements(session: nox.Session) -> None:
     """
-    Regenerate the pinned requirements files used in CI.
-
-    This session uses `uv pip compile` to regenerate the pinned
-    requirements files in `ci_requirements/` for use by the Nox sessions
-    for running tests, building documentation, and performing other
-    continuous integration checks.
+    Verify that the requirements in :file:`uv.lock` are compatible
+    with the requirements in `pyproject.toml`.
     """
-
-    session.install("uv")
-
-    category_version_resolution: list[tuple[str, str, str]] = [
-        ("tests", version, "highest") for version in supported_python_versions
-    ]
-
-    category_version_resolution += [
-        ("tests", minpython, "lowest-direct"),
-        ("docs", maxpython, "highest"),
-        ("all", maxpython, "highest"),
-    ]
-
-    category_flags: dict[str, tuple[str, ...]] = {
-        "all": ("--all-extras",),
-        "docs": ("--extra", "docs"),
-        "tests": ("--extra", "tests"),
-    }
```

Comment:
Dependency information across operating systems, versions of Python, and dependency sets is handled internally by `uv`, so these sections of the code are no longer needed.